apps/src/gamelab/reducers.js

```diff
@@ -48,8 +50,60 @@ function level(state, action) {
   }
 }
 
+function animation(state, action) {
+  state = state || { key: utils.createUuid() };
+
+  switch (action.type) {
+    case ActionType.SET_ANIMATION_NAME:
+      if (state.key === action.animationKey) {
+        return _.assign({}, state, {
+          name: action.name
+        });
+      }
+      return state;
+
+    default:
+      return state;
+  }
+}
+
+function animations(state, action) {
+  state = state || [];
+  
+  switch (action.type) {
+
+    case ActionType.ADD_ANIMATION_AT:
+      return [].concat(
+          state.slice(0, action.index),
+          action.animationProps,
+          state.slice(action.index));
+
+    case ActionType.DELETE_ANIMATION:
+      return state.filter(function (animation) {
+        return animation.key !== action.animationKey;
+      });
+
+    case ActionType.SET_INITIAL_ANIMATION_METADATA:
+      // ONLY FOR TESTING - REMOVE BEFORE SHIP!
```

Comment:
Not sure what "ship" is considered in regards to this comment. Worth adding a pivotal item to make sure it's not forgotten, or is this something that will be obvious enough that it's not really a concern.